src/output.cpp

```diff
@@ -1841,10 +1841,17 @@ get_bar( float cur, float max, int width, bool extra_resolution,
     status = status < 0 ? 0 : status;
     float sw = status * width;
 
-    nc_color col = colors[static_cast<int>( ( 1 - status ) * colors.size() )];
-    if( status == 0 ) {
-        col = colors.back();
-    } else if( ( sw < 0.5 ) && ( sw > 0 ) ) {
+    nc_color col;
+    if( !std::isfinite( status ) || colors.empty() ) {
+        col = c_red_red;
+    } else {
+        int ind = static_cast<int>( ( 1 - status ) * colors.size() );
+        ind = clamp<int>( ind, 0, colors.size() - 1 );
```

Comment:
It fails to compile on `gcc 5`:

```
11:15:51 In file included from src/ui_manager.h:7:0,
11:15:51                  from src/output.cpp:37:
11:15:51 src/cuboid_rectangle.h: In instantiation of ‘struct rectangle<int>’:
11:15:51 src/output.cpp:1849:53:   recursively required by substitution of ‘template<class Point> Point clamp(const Point&, const inclusive_rectangle<Point>&) [with Point = int]’
11:15:51 src/output.cpp:1849:53:   required from here
11:15:51 src/cuboid_rectangle.h:9:37: error: ‘dimension’ is not a member of ‘int’
11:15:51      static_assert( Point::dimension == 2, "rectangle is for 2D points; use cuboid for 3D points" );
11:15:51                                      ^
11:15:51 cc1plus: error: unrecognized command line option ‘-Wno-unknown-warning-option’ [-Werror]
```

You probably should not use `<int>` here.

Reply:
I'm planning to address this in a different way at #44598. Disallowing using a valid syntax just to support an old compiler is too restrictive in my opinion.